src/components/NewsletterSignup/NewsletterSignup.stories.tsx

```diff
@@ -0,0 +1,141 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import { withDesign } from "storybook-addon-designs";
+import useStateWithDependencies from "../../hooks/useStateWithDependencies";
+import NewsletterSignup from "./NewsletterSignup";
+
+const meta: Meta<typeof NewsletterSignup> = {
+  title: "Components/Form Elements/NewsletterSignup",
+  component: NewsletterSignup,
+  decorators: [withDesign],
+  parameters: {
+    jest: ["NewsletterSignup.test.tsx"],
+  },
+  argTypes: {
+    className: { control: false },
+    id: { control: false },
+    confirmationText: {
+      control: "text",
+      table: {
+        defaultValue: {
+          summary:
+            "Thank you! You have successfully subscribed to our email updates! You can update your email subscription preferences at any time using the links at the bottom of the email.",
+        },
+      },
+    },
+    descriptionText: {
+      control: "text",
+      table: {
+        defaultValue: {
+          summary:
+            "Stay connected with the latest research news from NYPL, including information about our events, programs, exhibitions, and collections.",
+        },
+      },
+    },
+    formHelperText: {
+      control: "text",
+    },
+    newsletterSignupType: {
+      control: "select",
+      table: {
+        defaultValue: {
+          summary: "whatsOn",
+        },
+      },
+    },
+    view: {
+      control: "select",
+      table: {
+        defaultValue: {
+          summary: "form",
+        },
+      },
+    },
+    title: {
+      control: "text",
+      table: { defaultValue: { summary: "Sign Up for Our Newsletter!" } },
+    },
+  },
+};
+
+export default meta;
+type Story = StoryObj<typeof NewsletterSignup>;
+
+const NewsletterSignupWithControls = (args) => {
+  const [view, setView] = useStateWithDependencies(args.view);
+  const [isInvalidEmail, setIsInvalidEmail] = useStateWithDependencies(
+    args.isInvalidEmail
+  );
+  const [confirmationText, setConfirmationText] = useStateWithDependencies(
+    args.confirmationText
+  );
+  const onSubmit = (values?: { [key: string]: string }) => {
+    let timer = setTimeout(() => {
+      setView("confirmation");
+      setConfirmationText(
+        "This is going to change your life. Check out those values in the console!"
+      );
+    }, 3000);
+    switch (values.email) {
+      case "":
+        setIsInvalidEmail(true);
+        clearTimeout(timer);
+        break;
+      case "error@nypl.org":
+        setView("error");
+        clearTimeout(timer);
+        break;
+    }
+    console.log(
+      "Submitted values:",
+      values,
+      "isInvalidEmail: ",
+      isInvalidEmail,
+      "View: ",
+      view
+    );
+  };
+  return (
+    <NewsletterSignup
+      {...args} // @todo All the same values below are already contained in this ...args array. But the ones below get to "win." I dunno why. Cuz they come after?
```

Comment:
Okay, _now_ I understand your concern about this. It does seem ordering matters when using the spread operator and yes it's because it comes after. I think I misunderstood at first.

Yes, `args` does contain the same values as the ones below but the values in `args` comes from the Storybook Controls section which the user can update. In this case, these values seem reasonable to have to override. For `confirmationText`, you don't need to have state for that since it seems it only is really updated once.

Reply:
Understood.